tests/system/Database/Live/ForgeTest.php

```diff
@@ -1137,6 +1139,70 @@ public function testCompositeKey()
         $this->forge->dropTable('forge_test_1', true);
     }
 
+    public function testSetKeyNames()
+    {
+        $this->forge->dropTable('forge_test_1', true);
+
+        $this->forge->addField([
+            'id' => [
+                'type'           => 'INTEGER',
+                'constraint'     => 3,
+                'auto_increment' => true,
+            ],
+            'code' => [
+                'type'       => 'VARCHAR',
+                'constraint' => 40,
+            ],
+            'company' => [
+                'type'       => 'VARCHAR',
+                'constraint' => 40,
+            ],
+            'active' => [
+                'type'       => 'INTEGER',
+                'constraint' => 1,
+            ],
+        ]);
+
+        $pk          = 'my_custom_pk';
+        $index       = 'my_custom_index';
+        $uniqueIndex = 'my_custom_unique_index';
+
+        if ($this->db->DBDriver === 'MySQLi' || $this->db->DBDriver === 'SQLite3') {
+            $pk = 'PRIMARY';
+        }
```

Comment:
If I remove the `if`, the assertion ` $this->assertSame($keys[$pk]->name, $pk)` fails in MySQLi/SQLite3.
But PRIMARY key is created.

`$pk` value is just ignored in MySQLi/SQLite3?
If `$pk` cannot be specified, it should either raise an error or be mentioned in the user guide.

Reply:
SQLite doesn't support dropping a primary key. So Codeigniter simply recreates the table without the primary key. Its name is meaningless. In an autoincrement PK CI generates a pseudo index named PRIMARY. 

MySQL supports naming a PK. The thing is though is that to drop it the command is DROP PRIMARY KEY. So the name again becomes meaningless. I just tested and even though you can specify the name when you do a SHOW INDEX the key name comes back as PRIMARY.

Reply:
I mean when creating keys.
If the `$pk` is meaningless in MySQL/SQLite3, it is better to add explanation in the docs.

```php
$this->forge->addPrimaryKey('id', $pk);
```

Reply:
Made a note